documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,272 @@
+
+```
+/subscriptions/830afc1a-b2c2-49fa-b36d-c05c335ea986/resourceGroups/MyResourceGroup/Microsoft.Compute/virtualMachines/MyVM/extensions/MyExtension
+```
+
+Because a resource's ID contains the identifier properties of the resource (resource group, name, parent name, etc.), we suggest that cmdlets have a parameter set that contains a `-ResourceId` parameter to allow the user to pass in the ID of the resource that they would like to act upon. Within the cmdlet code, the resource ID can be parsed to pull out these identifier properties and then pass them to the corresponding SDK call.
+
+The cmdlets currently found in the `Az.Resources` module that act upon resources are the following:
+
+- [`Get-AzResource`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzResource.md)
+    - Retrieves all resources in the current subscription or specified resource group, or gets a specific resource. The user can also filter results by resource types and tags on the given resources.
+- [`Move-AzResource`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Move-AzResource.md)
+    - Moves a resource to a different resource group or subscription.
+- [`New-AzResource`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzResource.md)
+    - Creates a resource with the given identifier properties and resource-specific properties (found in `-Properties`).
+- [`Remove-AzResource`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Remove-AzResource.md)
+    - Deletes the resource with the given identifier properties.
+- [`Set-AzResource`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Set-AzResource.md)
+    - Modifies a resource to have exactly the properties specified by the provided parameters (`PUT` operation).
+
+#### Resource Groups
+
+From the [_Azure Resource Manager overview_](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview#terminology) document, a resource group is defined as the following:
+
+_A container that holds related resources for an Azure solution. The resource group includes those resources that you want to manage as a group. You decide which resources belong in a resource group based on what makes the most sense for your organization._
+
+The cmdlets currently found in the `Az.Resources` module that act upon resource groups are the following:
+
+- [`Export-AzResourceGroup`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Export-AzResourceGroup.md)
+    - Converts the contents of a resource group to a template and saves it to a file on the user's machine.
+- [`Get-AzResourceGroup`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzResourceGroup.md)
```

Comment:
Can we expand here a bit on the implementation of the Get/New/Set Resource cmdlets, how they are implemented, how piping works, and current known issues?

Reply:
@markcowl updated -- talked about the implementation of the three cmdlets, whether they are using the `HttpClient` or `ResourceManagerClient` (or both), some piping examples, and notes about what we can look into in the future. Let me know if I should expand anymore.